OpenKeychain/src/main/java/org/sufficientlysecure/keychain/securitytoken/operations/PsoDecryptTokenOp.java

```diff
@@ -161,14 +161,23 @@ The shared secret obtained is a KEK (Key Encryption Key) that is used to wrap th
        */
         byte[] keyEncryptionKey = response.getData();
 
-        int xLen;
-        if (eckf.isX25519()) {
-            xLen = keyEncryptionKey.length;
-        } else {
+        /* From rfc6637#section-7 :
+        The input of KDF should be the x portion of the point.
+        As the result of ECDH can be expressed in two formats: compressed and uncompressed,
+        we have to deal with each case.
+        */
+        int xLen, startPos;
+        if (keyEncryptionKey[0] == 0x04 && keyEncryptionKey.length % 2 == 1) {
```

Comment:
do we really need ``&& keyEncryptionKey.length % 2 == 1``? Shouldn't 0x04 be sufficient?

Reply:
An uncompressed point is encoded as `04 || x || y`, with `x` and `y` are of the same size. However, a valid `x` may be led with `04`, so 0x04 is not sufficient.

Reply:
can you add this as a comment?

Reply:
Sure, I've added the description.